packages/cli/test/scripts/update.test.js

```diff
@@ -71,142 +74,188 @@ describe('update script', function() {
   const createProxies = async function() {
     this.networkFile = new NetworkFile(this.projectFile, network);
 
-    const contractsData = [
-      { name: 'ImplV1', alias: 'Impl' },
-      { name: 'WithLibraryImplV1', alias: 'WithLibraryImpl' },
-    ];
-    await add({ contractsData, projectFile: this.projectFile });
+    const contracts = ['ImplV1', 'WithLibraryImplV1'];
+    await add({ contracts, projectFile: this.projectFile });
     await push({ network, txParams, networkFile: this.networkFile });
 
-    this.implV1Address = this.networkFile.contract('Impl').address;
-    this.withLibraryImplV1Address = this.networkFile.contract('WithLibraryImpl').address;
+    this.implV1Address = this.networkFile.contract('ImplV1').address;
+    this.withLibraryImplV1Address = this.networkFile.contract('WithLibraryImplV1').address;
 
     this.proxy1 = await createProxy({
-      contractAlias: 'Impl',
+      contractName: 'ImplV1',
       network,
       txParams,
       networkFile: this.networkFile,
     });
     this.proxy2 = await createProxy({
-      contractAlias: 'Impl',
+      contractName: 'ImplV1',
       network,
       txParams,
       networkFile: this.networkFile,
     });
     this.proxy3 = await createProxy({
-      contractAlias: 'WithLibraryImpl',
+      contractName: 'WithLibraryImplV1',
       network,
       txParams,
       networkFile: this.networkFile,
     });
   };
 
-  const bumpVersion = async function() {
+  const bumpVersionAndPush = async function() {
+    const contracts = ['ImplV1', 'WithLibraryImplV1', 'ImplV2', 'WithLibraryImplV2'];
     await bump({ version: version2, txParams, projectFile: this.projectFile });
-    const newContractsData = [
-      { name: 'ImplV2', alias: 'Impl' },
-      { name: 'WithLibraryImplV2', alias: 'WithLibraryImpl' },
-    ];
     await add({
-      contractsData: newContractsData,
+      contracts,
       projectFile: this.projectFile,
     });
-    await push({ network, txParams, networkFile: this.networkFile });
 
-    this.implV2Address = this.networkFile.contract('Impl').address;
-    this.withLibraryImplV2Address = this.networkFile.contract('WithLibraryImpl').address;
+    await push({ contracts, network, txParams, networkFile: this.networkFile });
+
+    this.implV2Address = this.networkFile.contract('ImplV2').address;
+    this.withLibraryImplV2Address = this.networkFile.contract('WithLibraryImplV2').address;
+  };
+
+  const stubUpdate = (mappingToNames, mappingToAddresses) => {
```

Comment:
This approach is cool. I don't know if it will work 100% though.

For example, `getContractClass` probably returns something that contains the actual name, rather than the faked one, and I don't know if this can cause issues. On the other hand, this is very similar to what aliases did, so if aliases worked I guess this will work too.

I just think we should keep an eye on this in case it may break, and eventually think of a different way to test upgrades.

Reply:
Since these are unit tests we have to mock their dependencies. The only question which dependencies and how we mock. I've tried a few options before arriving at this set of mocks.